### DISCUSSION TITLE
ArgoCD Release Template

### DISCUSSION THREAD

**Discussion by adecchi-2inno** [helmfile/helmfile#246] [Questions]
I have setup a release template for Argo Application and then call it at each release, like the following example:

`  
   - name: grafana
    <<: *argocd-apps
    namespace: monitoring
    createNamespace: true
    disableValidation: true
    missingFileHandler: Warn
    labels:
      release-name: "{{ `{{ .Release.Name }}` }}"
      layer: "infra"
    set:
      - name: source.helm.file
        file: "helmfiles/grafana/{{ .Environment.Name }}/values.yaml.gotmpl"
`

So, my doubt here is to try to understand if possible to template a values.yaml.gotmpl used at **set** section previous to submit it to Argo
My intention is to template the values.yaml used by the grafana helm-chart and create the Grafana Argo CD Application. My intention is to use environment variable(**{{ requiredEnv "ARGOCD_INGRESS_NAME" }}**) to then pass it the gotmpl to create the Grafana ingress. If not possible , is possible to use [envsubst](envsubst (https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html) )  or another better way to work with it ?

**Comment by jouve** [comment]
hi,

you want to use [values](https://helmfile.readthedocs.io/en/latest/#values-files-templates) instead of `set` to use values.yaml template files

**Reply by adecchi-2inno** [reply]
I use **set** to pass the file path to the Template who is invoked, if in place of use **set** I use **values** works similar for the view of the Template ?

**Reply by yxxhero** [reply]
@adecchi-2inno could you show the expected? And format your code. thanks very much.
